### PR TITLE
Remove several old Agents

### LINKED ISSUE #264: Remove old/unused Agents
A while ago on a call we identified Agents that have had their functionality replaced and are no longer needed. We should remove them so that we don't need to maintain them (particularly the smurf_recorder, which has tests that depend on so3g.)

I believe these were:
- `agents/smurf_recorder/`
- `agents/pysmurf_archiver/`

I think we could also remove:
- `agents/hwp_sim/`
- `agents/smurf/`

The `smurf_stream_simulator` was used with the smurf-recorder. Is it still useful for other purposes? (My suspicion is yes, but I thought I'd ask.)

### PR DESCRIPTION
**Pull request by BrianJKoopman**
## Description

This removes the following agents:
- `smurf_recorder/` - Replaced by combination smurf-streamer + suprsync Agent
- `smurf/` - Very old directory that was starting point for smurf agents
- `hwp_sim/` - Unused early development example agent

## Motivation and Context

This removes what I think we can at the moment as discussed in #264.

Resolves #264.

## How Has This Been Tested?

I've run the tests locally and they've run successfully in the CI pipeline.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code cleanup/removal

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] Unless I am preparing a release, I have opened this PR onto the `develop` branch.

### CODE REVIEW

File: agents/ocs_plugin_so.py [hunk 1]
```diff
@@ -13,9 +13,8 @@
         ('Lakeshore370Agent', 'lakeshore370/LS370_agent.py'),
         ('Lakeshore240Agent', 'lakeshore240/LS240_agent.py'),
         ('Keithley2230G-PSU', 'keithley2230G-psu/keithley_agent.py'),
-        ('PysmurfController', 'smurf/pysmurf_control.py'),
+        ('PysmurfController', 'pysmurf_controller/pysmurf_controller.py'),
```
Comment (jlashner) [inline agents/ocs_plugin_so.py:16]: Question about this, does this  determine what host-manager runs? I haven't been touching this file much but I guess I should.

File: docs/agents/pysmurf/index.rst [hunk 2]
```diff
@@ -6,19 +6,17 @@
...
 
-These pages will show you how to setup these three agents.
+These pages will show you how to setup these two agents.
 
 .. toctree::
     :caption: Pysmurf agents:
     :maxdepth: 3
 
     pysmurf-controller
     pysmurf-monitor
```
Comment (jlashner) [inline docs/agents/pysmurf/index.rst:22]: I don't know how much this agent sub-section is helping. I kind of think we should just put the pysmurf-controller and monitor at the top-level in the docs. It's also kinda annoying to have to click `Pysmurf/OCS` on the sidebar before the actual agent docs.

**Review by jlashner — APPROVED** [review]
Hey Brian, looks good to me! One comment inline that you can change if you agree, but I'm happy merging this.

File: agents/ocs_plugin_so.py (hunk 1 above)
Comment (BrianJKoopman) [reply agents/ocs_plugin_so.py:16]: That's right, yes. Host Manager makes a call to [`ocs.site_config.scan_for_agents()`](https://ocs.readthedocs.io/en/develop/api.html?highlight=scan_for_agents#ocs.site_config.scan_for_agents), which uses this plugin file.

File: docs/agents/pysmurf/index.rst (hunk 2 above)
Comment (BrianJKoopman) [reply docs/agents/pysmurf/index.rst]: Agreed, I've removed it in 03bce4d.

**Comment by BrianJKoopman** [comment]
Thanks for taking a look! I'll merge once latest checks finish (or first thing in the morning.)

**Pull request merged**